code/game/machinery/vending.dm

```diff
@@ -307,7 +307,10 @@
 		if(panel_open)
 			attack_hand(user)
 		return
-	else if(istype(W, /obj/item/weapon/coin) && premium.len > 0)
+	else if(istype(W, /obj/item/weapon/coin))
+		if(!(premium.len > 0))

```

Comment:
I was gonna ask if that would still work with a negative value, but if a len is negative something is horribly wrong.